src/validators/OidcRecoveryValidator.sol

```diff
@@ -89,6 +93,43 @@ contract OidcRecoveryValidator is VerifierCaller, IModuleValidator, Initializabl
     return isNew;
   }
 
+  function startRecovery(StartRecoveryData calldata data, address targetAccount) external {
+    OidcKeyRegistry keyRegistryContract = OidcKeyRegistry(keyRegistry);
+    Groth16Verifier verifierContract = Groth16Verifier(verifier);
+
+    OidcData memory oidcData = accountData[targetAccount];
+    OidcKeyRegistry.Key memory key = keyRegistryContract.getKey(data.issHash, data.kid);
+
+    bytes32 senderHash = keccak256(abi.encode(msg.sender, oidcData.recoverNonce));
+
+    // Fill public inputs
+    uint8 index = 0;
```

Comment:
I would love to extract the pub inputs recreation to an internal function called something like `buildPublicInput`. That would make this logic a lot cleaner.